File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.js
/**
 * @param {number} n
 * @return {boolean}
 */
var checkDivisibility = function (n) {
    let sum = 0;
    let product = 1;
    let nclone = n;

    while (nclone > 0) {
        sum += Math.floor(nclone % 10);
        product *= Math.floor(nclone % 10);
        nclone = Math.floor(nclone / 10);
    }

    return (n % (sum + product)) === 0;
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        m = n
        sum = 0
        product = 1

        while m > 0:
            sum += m % 10
            product *= m % 10
            m = m // 10

        return (n % (sum + product)) == 0
